src/EPR.RegulatorService.Frontend.Web/Controllers/RegistrationSubmissions/RegistrationSubmissionsController.cs

```diff
@@ -115,11 +115,17 @@ public async Task<IActionResult> RegistrationSubmissionDetails(Guid? submissionI
     {
         _currentSession = await _sessionManager.GetSessionAsync(HttpContext.Session);
 
-        if (!GetAndRememberSubmissionDetails(submissionId, out var model))
+        var model = submissionId == null
+            ? (RegistrationSubmissionDetailsViewModel)_currentSession.RegulatorRegistrationSubmissionSession.SelectedRegistration
+            : (RegistrationSubmissionDetailsViewModel)await _facadeService.GetRegistrationSubmissionDetails(submissionId.Value);
```

Comment:
why is this casting?

Reply:
I think it's framework issue, if need this if define variable as var, But updated.